app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt

```diff
@@ -277,6 +280,48 @@ class WireNotificationManager @Inject constructor(
             }
     }
 
+    /**
+     * Infinitely listen for the screen that use is in
+     * and show notification for OngoingCall if user hides the app from the OngoingCallScreen.
+     * @param userIdFlow Flow of QualifiedID of User
+     * @param currentScreenState StateFlow that informs which screen is currently visible,
+     */
+    private suspend fun observeOngoingCallNotifications(
+        userIdFlow: Flow<UserId?>,
+        currentScreenState: StateFlow<CurrentScreen>
+    ) {
+        currentScreenState
+            .combine(userIdFlow, ::Pair)
+            .flatMapLatest { (currentScreen, userId) ->
+                if (currentScreen == CurrentScreen.InBackground && userId != null) {
+                    // user was in OngoingCallScreen and hides the app,
+                    // so we need to display OngoingCall notifications
+                    coreLogic.getSessionScope(userId)
+                        .calls
+                        .establishedCall()
+                        .map { calls ->
+                            // looking for the call that user was in
+                            val currentOngoingCall = calls.firstOrNull()
+
+                            // combining founded ongoingCall with a userId for displaying a notification
+                            Optional.ofNullable(currentOngoingCall) to userId
+                        }
+                } else {
+                    // do nothing if app is not in Background, or there is no user
```

Comment:
Should there be any incoming call if there is no user registered? 🤔

Reply:
obviously - no, but if user logged out we need to "stop listening" ongoingCalls, it's implemented by `flatmapLatest`